src/google/protobuf/compiler/objectivec/objectivec_file.cc

```diff
@@ -534,6 +534,12 @@ void FileGenerator::PrintFileRuntimePreamble(
       "\n",
       "filename", file_->name());
 
+  printer->Print(
+    "#if __has_feature(objc_arc)\n"
+    "  #error This file should be compiled with ARC. Use the -fobjc-arc Compiler Flag to disable ARC for this file.\n"
+    "#endif"
+    "\n");
+
```

Comment:
The string here is wrong, should say: "This file should **not** be compiled with ARC...".
This warning is also included in the header file which it should not be. :/